### ISSUE TITLE
Alternative on pageurl

### ISSUE THREAD

**Issue by Speedy1991** [wagtail/wagtail#4598]
I patched pageurl to take an fallback viewname if page is None.
Usecase: It's easy to implement comming soon pages or alternative pages if the page is not available (yet).  I don't know if this is useful for others.
Let me know if I should start a PR if you guys are interested.
```
@register.simple_tag(takes_context=True)
def pageurl(context, page, fallback=None):

    if not page and fallback:
        return reverse(fallback)

    try:
        current_site = context['request'].site
    except (KeyError, AttributeError):
        return page.url

    return page.relative_url(current_site, request=context.get('request'))
```
Example calls:
```
<li><a href="{% pageurl page.download_page 'coming_soon' %}">{% trans "Download" %}</a></li>
<li><a href="{% pageurl page.profile_page 'accounts:mock' %}">{% trans "Profile" %}</a></li>
```

**Comment by gasman** [comment]
Thanks @Speedy1991! This looks like a useful addition, and a PR would be very welcome.

I'd suggest that the `fallback` parameter should be a keyword argument rather than a positional one, e.g. `{% pageurl page.download_page fallback='coming_soon' %}` - so that it's more understandable to someone reading the template, and leaves open the possibility of us extending the `{% pageurl %}` tag in other ways in future. (Perhaps this works already with the existing code...)

**Comment by gasman** [comment]
Implemented in #4603
